chia/wallet/wallet_state_manager.py

```diff
@@ -651,10 +661,82 @@ async def determine_coin_type(
+                    clawback_coins.append((coin.coin, metadata))
+                    if len(clawback_coins) >= self.config.get("auto_claim_coin_size", 50):
+                        await self.claim_clawback_coins(clawback_coins, tx_fee)
+                        clawback_coins = []
+        if len(clawback_coins) > 0:
+            print(f"Auto claim fee {tx_fee}")
+            await self.claim_clawback_coins(clawback_coins, tx_fee)
+
+    async def claim_clawback_coins(self, merkle_coins: List[Tuple[Coin, Dict[str, Any]]], fee: uint64) -> List[bytes32]:
+        assert len(merkle_coins) > 0
+        coin_spends: List[CoinSpend] = []
+        message_list: List[bytes32] = [c[0].name() for c in merkle_coins]
+        message: bytes32 = std_hash(b"".join(message_list))
+        clawback_coins = []
+        for coin, metadata in merkle_coins:
+            recipient_puzhash = bytes32.from_hexstr(metadata["recipient_puzhash"])
+            sender_puzhash = bytes32.from_hexstr(metadata["sender_puzhash"])
+            if metadata["is_recipient"]:
+                derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(recipient_puzhash)
+            else:
+                derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(sender_puzhash)
+            if derivation_record is None:
+                self.log.warning(f"You are not able to spend coin {coin.name().hex()}")
+                continue
+            inner_puzzle: Program = self.main_wallet.puzzle_for_pk(derivation_record.pubkey)
+            inner_solution: Program = self.main_wallet.make_solution(
+                primaries=[
+                    {
+                        "puzzlehash": recipient_puzhash if metadata["is_recipient"] else sender_puzhash,
+                        "amount": uint64(coin.amount),
+                        "memos": [],
+                    }
+                ],
+                coin_announcements=None if len(coin_spends) > 0 or fee == 0 else {message},
+            )
+            coin_spends.append(generate_clawback_spend_bundle(coin, metadata, inner_puzzle, inner_solution))
+            clawback_coins.append(coin.name())
+        spend_bundle = await self.main_wallet.sign_transaction(coin_spends)
+        if fee > 0:
+            chia_tx = await self.main_wallet.create_tandem_xch_tx(
+                fee, Announcement(coin_spends[0].coin.name(), message)
+            )
+            assert chia_tx.spend_bundle is not None
+            spend_bundle = SpendBundle.aggregate([spend_bundle, chia_tx.spend_bundle])
+            chia_tx = dataclasses.replace(chia_tx, spend_bundle=None)
+            await self.add_pending_transaction(chia_tx)
+        # We don't want to mess up tx record here so just push the bundle
+        await self.wallet_node.push_tx(spend_bundle)
+        for coin_spend in coin_spends:
+            # This will make sure we don't double spend
```

Comment:
For the wallet, the mempool is not trustable unless we own the fullnode. So far there is a method to access the mempool, I need to add more things to support this. If the database change is a concern, I suggest we use cache to prevent unnecessary double-spending.

Reply:
Oh yeah sure we are on the wallet side here.. right so then either add a cache for that or maybe we could check `WalletTransactionStore.get_all_unconfirmed` to see if the coin is in one of the spend bundles there? Little more heavy but since it not happens very often thats probably acceptable? Of course that would require the tx record including the spend bundle, what you currently don't add for some reason (you only add the fee tx). Maybe you can elaborate on this too :)